### PR TITLE
#3072 - Body and headers on error responses

### LINKED ISSUE #3072: 403 error don't show Response header and body
I used the current swagger UI running on http://petstore.swagger.io/

I noticed that ``40*`` errors do not show the response header and body.
In my case I want to return the error code and error message as json object in the body like this:
```json
{
  "code": 403,
  "message": "You are not authorized!"
}
````
Even if the body is set correctly and content-type is set, the body/header are not shown in the current version. 

image

### PR DESCRIPTION
**Pull request by owenconti**
Fixes #3072 - I would appreciate someone else testing this, as I'm not sure I know how to trigger various different errors. I tested mostly with a 404 response.

Added a test to ensure the error and response objects are merged correctly.

### CODE REVIEW

File: src/core/plugins/spec/reducers.js [hunk 1]
```diff
@@ -86,7 +91,7 @@ export default {
     let newState = state.setIn( [ "responses", path, method ], fromJSOrdered(result) )
 
     // ImmutableJS messes up Blob. Needs to reset its value.
-    if (res.data instanceof win.Blob) {
+    if (win.Blob && res.data instanceof win.Blob) {
```
Comment (owenconti) [inline src/core/plugins/spec/reducers.js]: Test suite was throwing an error:

`Right-hand side of 'instanceof' is not an object`

**Comment by webron** [comment]
It's going to be hard to test all use cases, but indeed the 404 works as expected.

@owenconti I believe this also covers the second half of the issue described in #3041?

**Comment by owenconti** [comment]
@webron I think so. Here is a screenshot using the bad data as provided from that issue:

screen shot 2017-06-26 at 4 26 50 pm

**Comment by webron** [comment]
Okay, great.

**Pull request merged**